test_harness/src/lib.rs

```diff
@@ -2308,6 +2775,49 @@ mod tests {
             .send()
             .await?)
     }
+
+    async fn get_analytics(
+        api_client: &Client,
+        url: &ApiUrl,
+        query: &AnalyticsQuery,
+        token: Option<String>,
+    ) -> anyhow::Result<Vec<FetchedAnalytics>> {
+        let query = serde_qs::to_string(query).expect("should parse query");
+
+        let analytics = match token {
+            Some(token) => {
+                let endpoint_url = url
+                    .join(&format!("v5/analytics/for-advertiser?{}", query))
+                    .expect("valid endpoint");
+
+                api_client
+                    .get(endpoint_url)
+                    .bearer_auth(&token)
+                    .send()
+                    .await
+                    .expect("failed to get analytics")
+                    .json::<Vec<FetchedAnalytics>>()
+                    .await
+                    .expect("failed to get json")
+            }
```

Comment:
can we simplify this case by just setting the Auth token for the `/analytics` request instead of calling `for-advertiser`?
It will filter the results accordingly to the token either way.

We can always add additional test for /for-advertiser & for-publisher eventually but that's an overkill for now